core/store/src/trie/state_snapshot.rs

```diff
@@ -153,52 +153,39 @@ impl ShardTries {
     }
 
     /// Makes a snapshot of the current state of the DB.
-    /// If a snapshot was previously available, it gets deleted.
-    pub fn make_state_snapshot(
+    pub fn create_state_snapshot(
         &self,
-        prev_block_hash: &CryptoHash,
+        prev_block_hash: CryptoHash,
         shard_uids: &[ShardUId],
         block: &Block,
     ) -> Result<(), anyhow::Error> {
         metrics::HAS_STATE_SNAPSHOT.set(0);
         // The function returns an `anyhow::Error`, because no special handling of errors is done yet. The errors are logged and ignored.
         let _span =
-            tracing::info_span!(target: "state_snapshot", "make_state_snapshot", ?prev_block_hash)
+            tracing::info_span!(target: "state_snapshot", "create_state_snapshot", ?prev_block_hash)
                 .entered();
-        tracing::info!(target: "state_snapshot", ?prev_block_hash, "make_state_snapshot");
-
-        let StateSnapshotConfig { home_dir, hot_store_path, state_snapshot_subdir, .. } =
-            self.state_snapshot_config();
+        let _timer = metrics::CREATE_STATE_SNAPSHOT_ELAPSED.start_timer();
 
-        let _timer = metrics::MAKE_STATE_SNAPSHOT_ELAPSED.start_timer();
         // `write()` lock is held for the whole duration of this function.
-        // Accessing the snapshot in other parts of the system will fail.
-        let mut state_snapshot_lock = self.state_snapshot().write().map_err(|err| {
-            anyhow::Error::msg(format!(
-                "error accessing write lock of state_snapshot: {}",
-                err.to_string()
-            ))
-        })?;
+        let mut state_snapshot_lock = self.state_snapshot().write().unwrap();
```

Comment:
Did some digging into poison lock error, and it basically happens when a thread panics holding the lock. While it's not a good pattern to use unwrap in rust, online articles suggest it's fine in this use case as we are effectively propagating the error from a different thread that had panicked holding the lock.

In our case, we really don't expect that to ever happen...